src/Microsoft.DotNet.Tools.Compiler.Native/ArgValues.cs

```diff
@@ -56,8 +57,20 @@ public NativeCompileSettings GetNativeCompileSettings()
 
             if (!string.IsNullOrEmpty(IlcPath))
             {
-                config.IlcPath = IlcPath;
-                config.IlcSdkPath = IlcPath;
+                // We want a directory path. If the user gave us the exact path to the executable
```

Comment:
I took some time to think over this, and agree with what you are saying about the properties. Indeed that lack of roundtrip seems bad.

I don't see ArgValues going away in the near future, and while it exists it is the best place to centralize mutating arguments before they are set in NativeCompileSettings. Thanks :smile: